assets/tests/components/detours/detourMap.test.tsx

```diff
@@ -224,4 +225,44 @@ describe("DetourMap", () => {
       container.querySelector("img[src^=test_satellite_url")
     ).not.toBeNull()
   })
+
+  test("stops are drawn on the map", () => {
+    const stops = [
+      { ...stopFactory.build(), missed: false },
+      { ...stopFactory.build(), missed: false },
+      { ...stopFactory.build(), missed: false },
+    ]
+
+    const { container } = render(<DetourMapWithDefaults stops={stops} />)
+
+    expect(container.querySelectorAll(".c-stop-icon")).toHaveLength(3)
+  })
+
+  test("stops are drawn as missed stop icons when they are missed", () => {
+    const stops = [
+      { ...stopFactory.build(), missed: true },
+      { ...stopFactory.build(), missed: true },
+      { ...stopFactory.build(), missed: false },
+    ]
+
+    const { container } = render(<DetourMapWithDefaults stops={stops} />)
+
+    expect(container.querySelectorAll(".c-stop-icon")).toHaveLength(1)
+    expect(container.querySelectorAll(".c-missed-stop-icon")).toHaveLength(2)
+  })
+
+  test("duplicate stops are only drawn once", () => {
+    const duplicateStop = { ...stopFactory.build(), missed: true }
+    const stops = [
+      { ...stopFactory.build(), missed: true },
+      duplicateStop,
+      { ...stopFactory.build(), missed: false },
+      duplicateStop,
+    ]
+
+    const { container } = render(<DetourMapWithDefaults stops={stops} />)
+
+    expect(container.querySelectorAll(".c-stop-icon")).toHaveLength(1)
```

Comment:
suggestion: what do you think about wrapping this and the missed stop icon selectors up as a reusable query (`stopIcons.get`?) like https://github.com/mbta/skate/blob/e8905231c87f0d7ab712c1879eb244ac25c6734b/assets/tests/testHelpers/selectors/components/mapPage/map.ts#L10-L15 and putting it next to `stopIcon.get` in https://github.com/mbta/skate/blob/e8905231c87f0d7ab712c1879eb244ac25c6734b/assets/tests/testHelpers/selectors/components/map/markers/stopIcon.ts#L4-L11 ?

Reply:
Oh! I just read the return value from [`byRole`](https://github.com/domasx2/testing-library-selector) and I think it'd be better to add this as `stopIcon.getAll`, not `stopIcons.get()`

Reply:
Love it!

Would you recommend doing missed stops as `stopIcon.getAllMissed()` or as `missedStopIcon.getAll()`?

Reply:
I'd prefer `missedStopIcon.getAll()`, in the case that we can replace this with a  selector from the `byRole` library in the future, the API would be consistent if we did it this way IMO

Reply:
Agreed - it also _feels_ more _right_ to do `missedStopIcon.getAll()`.

Reply:
[This is a lot nicer](https://github.com/mbta/skate/pull/2484/commits/332c63ee924269f3704ee5ececbd597eaf741659)